lib/matplotlib/testing/decorators.py

```diff
@@ -389,14 +389,14 @@ def decorator(func):
 
         @pytest.mark.parametrize("ext", extensions)
         def wrapper(*args, ext, **kwargs):
+            fn = getattr(wrapper, "_item_name_", func.__name__)
```

Comment:
Can you instead insert `request` as additional parameter/fixture to the function (https://docs.pytest.org/en/latest/reference.html#request)?  Then the item name should be available as `request.node.name` (always, you don't even need the getattr).

This would make the thing work even without pytest_pyfunc_call (which is relevant to let third-parties want to use check_figures_equal without having to copy our entire pytest config).

Reply:
Ah, great!  I suspected there was an easier way but was failing at finding what I wanted in their documentation.

Reply:
you may also want to asciisafe/punycode/urlencode/whatever the name, just to be safe with the fs on windows.

Reply:
Force-pushed out of existence.

Reply:
At the very least, you should replace `[/\\]` because it won't be able to write to a non-existent subdirectory.